util/libcrypto.num

```diff
@@ -5425,3 +5425,4 @@ ASN1_item_d2i_ex                        5552	3_0_0	EXIST::FUNCTION:
 ASN1_TIME_print_ex                      5553	3_0_0	EXIST::FUNCTION:
 EVP_PKEY_get0_provider                  5554	3_0_0	EXIST::FUNCTION:
 EVP_PKEY_CTX_get0_provider              5555	3_0_0	EXIST::FUNCTION:
+err_clear_last_constant_time            5556  3_0_0 EXIST::FUNCTION:
```

Comment:
Why is this line added?

Reply:
The `libcommon` target would not be able to resolve the `err_clear_last_constant_time` symbol without this.

Reply:
Drive by comment: It seems strange that we need to add an internal symbol to libcrypto.num....but aside from that (assuming that we do need to do this), I guess this should be 3_1_0 and not 3_0_0 since we are targeting master.

Reply:
The symbol is used in the provider cipher implementations. It would not be resolved otherwise. Maybe the problem is that err_clear_last_constant_time should not be internal to begin with?

Reply:
I still don't understand why this is essential.  Libcrypto should be linkable without it being public.  The FIPS provider accesses it indirectly.  The default provider can do either since it's built in....

Reply:
The symbol won't be resolved without it being public. I have already tried.

Reply:
> The symbol won't be resolved without it being public. I have already tried.

This file says what's going to be exposed in the *shared library libcrypto.so*, nothing else.
So that's not it.

This can't be caused by the built in providers (base and default), since they are built into libcrypto and symbol resolution should therefore not be a problem.

I suspect this is about the legacy provider, which is also normally built as a dynamically loadable module as well, so it's likely that if you duplicate your fipsprov.c changes to legacyprov.c as well, things will go smoother.

Reply:
I did exactly this in another branch, it builds fine but some tests fail. This is the only change:
```c
int OSSL_provider_init(const OSSL_CORE_HANDLE *handle,
                       const OSSL_DISPATCH *in,
                       const OSSL_DISPATCH **out,
                       void **provctx)
{
    OSSL_LIB_CTX *libctx = NULL;

    for (; in->function_id != 0; in++) {
        #define set_func(c, f) if (c == NULL) c = f; else if (c != f) return 0;

        switch (in->function_id) {
        case OSSL_FUNC_CORE_CLEAR_LAST_CONSTANT_TIME:
            set_func(c_clear_last_constant_time, OSSL_FUNC_core_clear_last_constant_time(in));
            break;
        default:
            break;
        }
    }

    if ((*provctx = ossl_prov_ctx_new()) == NULL
        || (libctx = OSSL_LIB_CTX_new_child(handle, in)) == NULL) {
        OSSL_LIB_CTX_free(libctx);
        legacy_teardown(*provctx);
        *provctx = NULL;
        return 0;
    }
    ossl_prov_ctx_set0_libctx(*provctx, libctx);
    ossl_prov_ctx_set0_handle(*provctx, handle);

    *out = legacy_dispatch_table;

    return 1;
}
```